src/read_data_phantom_hdf5.f90

```diff
@@ -0,0 +1,690 @@
+!-----------------------------------------------------------------
+!
+!  This file is (or was) part of SPLASH, a visualisation tool
+!  for Smoothed Particle Hydrodynamics written by Daniel Price:
+!
+!  http://users.monash.edu.au/~dprice/splash
+!
+!  SPLASH comes with ABSOLUTELY NO WARRANTY.
+!  This is free software; and you are welcome to redistribute
+!  it under the terms of the GNU General Public License
+!  (see LICENSE file for details) and the provision that
+!  this notice remains intact. If you modify this file, please
+!  note section 2a) of the GPLv2 states that:
+!
+!  a) You must cause the modified files to carry prominent notices
+!     stating that you changed the files and the date of any change.
+!
+!  Copyright (C) 2005-2017 Daniel Price. All rights reserved.
+!  Contact: daniel.price@monash.edu
+!
+!-----------------------------------------------------------------
+module readphantomhdf5
+
+  implicit none
+
+  integer, parameter :: itypemap_sink_phantom    = 3
+  integer, parameter :: itypemap_unknown_phantom = 9
+
+contains
+
+!---------------------------------------------------------------------
+! function mapping iphase setting in Phantom to splash particle types
+!---------------------------------------------------------------------
+elemental integer function itypemap_phantom(iphase)
+  integer(kind=1), intent(in) :: iphase
+
+  select case(int(iphase))
+  case(1:2)
+    itypemap_phantom = iphase
+  case(3:7) ! put sinks as type 3, everything else shifted by one
+    itypemap_phantom = iphase + 1
+  case(-3) ! sink particles, either from external_binary or read from dump
+    itypemap_phantom = itypemap_sink_phantom
+  case default
+    itypemap_phantom = itypemap_unknown_phantom
+  end select
+
+end function itypemap_phantom
+
+subroutine get_ncolumns(file_id,ncolumns,ndusttypes,ndustsmall,nptmass)
+  use hdf5_utils, only:HID_T,exist_in_hdf5,open_hdf5group,close_hdf5group
```

Comment:
nothing else in splash currently uses the fortran hdf5 library, this causes all kinds of dependency problems because one needs to match the versions of gfortran used to compile both splash and hdf5. This is the reason all hdf5 support in splash is implemented in c.